codersdk/deployment.go

```diff
@@ -2418,6 +2420,24 @@ func (c *DeploymentValues) Options() serpent.OptionSet {
 			Value:       serpent.EnumOf(&c.PostgresAuth, PostgresAuthDrivers...),
 			YAML:        "pgAuth",
 		},
+		{
+			Name:        "Postgres Max Connections",
+			Description: "The maximum number of connections to maintain in the connection pool. 0 is unlimited.",
```

Comment:
Is allowing a user to set an unbounded number of connections reasonable?

Reply:
I've added a warning and a discouragement for its use, but I don't think we should not allow this if operators deem it necessary.